docs/extending-cadl/basics.md

```diff
@@ -159,3 +159,96 @@ model Employee extends Person {
 ## Next steps
 
 Cadl libraries can contain more than just types. Read the subsequent topics for more details on how to write [decorators](./create-decorators.md), [emitters](./emitters-basics.md) and [linters](./linters.md).
+
+## Testing
+
+Cadl provides a testing framework to help testing libraries. Examples here are shown using `mocha` but any other JS test framework can be used.
+
+### Define the testing library
+
+First step is to define how your library can be loaded from the test framework. This will let your library to be reused by other library test.
+
+1. Create a new file `./src/testing/index.ts` with the following content
+
+```ts
+export const MyTestLibrary = createTestLibrary({
+  name: "<name-of-npm-pkg>",
+  packageRoot: resolvePath(fileURLToPath(import.meta.url), "../../../../"),
```

Comment:
Should document what this path is, depending on how your package is laid out you may need a different path here?